config/env.config.js

```diff
@@ -55,5 +57,77 @@ const config = {
+                    `
+                    <script>
+                    (function() {
+                        function getParameterByName(name) {
+                            var url = window.location.search;
+                            var queryExp = new RegExp("[\\?&]"+name+"=([^&#]*)", "i");
+                            var match = queryExp.exec(url);
+                            if (match && match.length > 1) {
+                                return match[1];
+                            }
+                            return null;
+                        };
+                        var originUrl = (getParameterByName("originEnvironment") || "").toLowerCase();
+                        var targetUrl = (getParameterByName("targetEnvironment") || "").toLowerCase();
+
+                        // Set target environment for origin environment to redirect to
+                        if (targetUrl.length > 0) {
+                            targetUrl = decodeURIComponent(targetUrl)
+                            // Clear origin environment's local storage if target = origin
+                            if (window.location.href.toLowerCase().indexOf(targetUrl) != -1) {
+                                window.localStorage.removeItem("${PLAYGROUND_REDIRECT}");
+                                return;
+                            }
+
+                            window.localStorage.setItem("${PLAYGROUND_REDIRECT}", targetUrl);
+                        }
+
+                        // Redirect origin environment to target
+                        var redirectUrl = window.localStorage.getItem("${PLAYGROUND_REDIRECT}");
+                        if (redirectUrl) {
+                            var originParam = [
+                                (window.location.search ? "&" : "?"), 
+                                "originEnvironment=",
+                                encodeURIComponent(window.location.origin)
+                            ].join("");
+
+                            window.location.href = [
+                                redirectUrl,
+                                window.location.pathname,
+                                window.location.search,
+                                originParam,
+                                window.location.hash
+                            ].join("");
+                        }
+
+                        // Point app environment back to origin if user is not in origin
+                        if (originUrl.length > 0) {
+                            window.localStorage.setItem("${PLAYGROUND_ORIGIN}", decodeURIComponent(originUrl).toLowerCase());
+                        }
+
+                        // If reached here, environment is already configured
+                    })();
+                    </script>
+                    ` + 
+                    htmlPluginData.html.slice(htmlHead + headOpeningTag.length);
+                }
+                callback(null, htmlPluginData);
```

Comment:
Do you mean for the `callback()` function? The code I used to create my personal `RedirectPlugin` is derived from the HTMLPLugin documentation:

https://github.com/jantimon/html-webpack-plugin